honeybee_radiance/lightsource/sky/skymatrix.py

```diff
@@ -0,0 +1,163 @@
+        yields a higher resolution sky using the Reinhart patch subdivision. For example,
+        setting density to 4 yields a sky with 2305 patches plus one patch for the
+        ground.
+        """
+        return self._density
+
+    @density.setter
+    def density(self, value):
+        typing.int_in_range(value, 1, input_name='SkyMatrix subdivision density')
+        self._density = value
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create the sky from a dictionary.
+
+        Args:
+            input_dict: A python dictionary in the following format
+
+        .. code-block:: python
+
+                {
+                'type': 'SkyMatrix',
+                'wea': {},
+                'north': 0.0  # optional
+                'density': 1  # optional
+                }
+        """
+        if 'type' not in input_dict or input_dict['type'] != 'SkyMatrix':
+            raise ValueError('Input dict "type" must be "SkyMatrix".')
+        if 'north' in input_dict:
+            north = input_dict['north']
+        else:
+            north = 0
+        if 'density' in input_dict:
+            density = input_dict['density']
+        else:
+            density = 1
+
+        sky = cls(Wea.from_dict(input_dict['wea']), north, density)
+
+        return sky
+
+    # TODO: add support for additional parameters
+    # TODO: add gendaymtx to radiance-command and use it for validating inputs
+    def to_radiance(
+            self, output_type=0, wea_file=None, output_name=None, cumulative=False,
+            components=0):
+        """Return Radiance command to generate the sky.
+
+        Note that you need to write the wea to a file (in.wea) before running this
+        command.
+
+        Alternatively you can use write method which will write the wea data to a file.
+
+        Args:
+            output_type: An integer between 0 to 1 for output type.
+                * 0 = output in W/m2/sr visible (default)
+                * 1 = output in W/m2/sr solar
+            wea_file: Path to wea file (default: in.wea).
+            output_name: A name for output files (default: sky_mtx).
+            cumulative: A boolean to generate cumulative sky. This option is only
+                available in Radiance 5.3 and higher versions (default: False).
+            components: An integer between 0-2 to note the distribution of which
+                components should be included. 0 might be used to include both solar and
+                sky contribution. 1 may be used to produce a sun-only matrix, with no sky
+                contributions.  Alternatively, 2 may be used to exclude any direct solar
+                component from the output.  If there is a sun in the description,
+                gendaymtx will include its contribution in the four nearest sky patches,
+                distributing energy according to centroid proximity (default: 0).
```

Comment:
I feel like some of the vocabulary here can be clearer or more consistent. Is your use of the word `solar` and `sun` interchangeable? Maybe it's just better to stick to using `sun`. It's also a little confusing that sometimes, you include the word `direct solar` but other times, you just say `solar`. Are these two the same?